src/fileformat/file_format/pe/authenticode/pkcs7_signature.cpp

```diff
@@ -556,18 +557,27 @@ std::vector<DigitalSignature> Pkcs7Signature::getSignatures(const retdec::filefo
 
 	signature.signer = signer;
 
+	// If there are warnings, then the signature is invalid
+	// We don't use the countersignature warnings here as
+	// previous implementation did when verifying
+	if (signature.warnings.empty()) {
```

Comment:
Can this be simplified to `signature.isValid = signature.warnings.empty();`. It also reads nice as a sentence

> Signature is valid whenever there are no warnings.

Reply:
Yes, that's much better